python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -626,6 +627,24 @@ def __set__(self, obj, _val):
         GangaObject.__setattr__(raw_obj, attr_name, final_val)
 
 
+def proxy_wrap(f):
+    # type: (Callable) -> Callable
+    """
+    A decorator to strip the proxy from all incoming arguments
+    (including ``self`` if it's a method) and add one to the return
+    value.
+    """
+
+    @functools.wraps(f)
+    def proxy_wrapped(*args, **kwargs):
+        s_args = [stripProxy(a) for a in args]
```

Comment:
As long as we're now stripping the proxies everywhere then OK I'm convinced, I'm just weary that users are usually good at finding corner cases such as the proxies which don't use the automatic format.